src/integrationTest/java/org/radarcns/webapp/SourceEndPointTest.java

```diff
@@ -49,6 +61,62 @@
 
     @Test
     public void getAllSourcesForSubjectInProject() throws IOException {
+        List<Source> sources = requestSourcesOfSubject();
+        // not effective time available
+        assertEquals(SourceStatus.UNKNOWN, sources.get(0).getStatus());
+    }
+
+    @Test
+    public void getAllSourcesForSubjectInProjectDisconnected() throws IOException {
+        Instant now = Instant.now();
+        insertMonitorStatistics(now.minus(Duration.ofDays(2)), now.minus(Duration.ofHours(20)));
+
+        List<Source> sources = requestSourcesOfSubject();
+        assertEquals(SourceStatus.DISCONNECTED, sources.get(0).getStatus());
+    }
+
+    @Test
+    public void getAllSourcesForSubjectInProjectConnected() throws IOException {
+        Instant now = Instant.now();
+        insertMonitorStatistics(now.minus(Duration.ofMinutes(30)), now);
+
+        // not effective time available
+        List<Source> sources = requestSourcesOfSubject();
+        assertEquals(SourceStatus.CONNECTED, sources.get(0).getStatus());
+    }
+
+    @Test
+    public void getSourceByIdConnected() throws IOException {
+        Instant now = Instant.now();
+        insertMonitorStatistics(now.minus(Duration.ofMinutes(30)), now);
+
+        String requestPath = PROJECTS + '/' + PROJECT + '/' + SUBJECTS + '/' + SUBJECT + '/'
+                + SOURCES + '/' + SOURCE;
+        try (Response response = apiClient.request(requestPath, APPLICATION_JSON, Status.OK)) {
+            assertNotNull(response);
+            assertTrue(response.isSuccessful());
+
+            ObjectReader reader = RadarConverter.readerFor(Source.class);
+            Source sources = reader.readValue(response.body().byteStream());
```

Comment:
Use `getJson(requestPath, Source.class, Status.OK)` here instead (from #84).

Reply:
I will merge it